From f7e9dc7781e4ccdb3ac172b96fa4d91c08a3a99c Mon Sep 17 00:00:00 2001
From: WangJing <wangjing@mprc.pku.edu.cn>
Date: Wed, 26 Oct 2011 17:04:15 +0800
Subject: [PATCH 032/641] UniCore64: Add pmd related functions

Fixed errors:
mm/built-in.o: In function `__pte_alloc':
(.text+0x17c98): undefined reference to `pmd_populate'
mm/built-in.o: In function `__pte_alloc_kernel':
(.text+0x17cf2): undefined reference to `pmd_populate_kernel'
mm/built-in.o: In function `__pmd_alloc':
(.text+0x18f1a): undefined reference to `pmd_alloc_one'
mm/built-in.o: In function `__pmd_alloc':
(.text+0x18f57): undefined reference to `pmd_free'

Signed-off-by: WangJing <wangjing@mprc.pku.edu.cn>
---
 arch/unicore64/mm/pgtable.c |   48 +++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/arch/unicore64/mm/pgtable.c b/arch/unicore64/mm/pgtable.c
index f538b76..32c10de 100644
--- a/arch/unicore64/mm/pgtable.c
+++ b/arch/unicore64/mm/pgtable.c
@@ -1,4 +1,6 @@
+#include <linux/kernel.h>
 #include <linux/mm.h>
+#include <asm/pgtable.h>
 
 /**
  * pgd_alloc() - Allocates a new Page Global Directory
@@ -102,3 +104,49 @@ void pud_populate(struct mm_struct *mm, pud_t *pudp, pmd_t *pmd)
 	/* FIXME */
 	BUG();
 }
+
+/**
+ * pmd_populate() -
+ * @mm:
+ * @pmd:
+ * @page:
+ */
+void pmd_populate(struct mm_struct *mm, pmd_t *pmd, pgtable_t page)
+{
+	/* FIXME */
+	BUG();
+}
+
+/**
+ * pmd_populate_kernel() -
+ * @mm:
+ * @pmd:
+ * @ptep:
+ */
+void pmd_populate_kernel(struct mm_struct *mm, pmd_t *pmd, pte_t *ptep)
+{
+	/* FIXME */
+	BUG();
+}
+
+/**
+ * pmd_alloc_one() -
+ * @mm:
+ * @addr:
+ */
+pmd_t *pmd_alloc_one(struct mm_struct *mm, unsigned long addr)
+{
+	/* FIXME */
+	BUG();
+}
+
+/**
+ * pmd_free() -
+ * @mm:
+ * @pmdp:
+ */
+void pmd_free(struct mm_struct *mm, pmd_t *pmdp)
+{
+	/* FIXME */
+	BUG();
+}
-- 
1.7.9.5

